packages/frontend/src/pages/Dashboard.tsx

```diff
@@ -102,6 +101,12 @@ const Dashboard = () => {
         }
     }, [dashboardUuid, history, isSuccess, projectUuid, reset]);
 
+    useEffect(() => {
+        if (isEditMode && haveFiltersChanged !== dashboardFilters) {
```

Comment:
Actually, checking the filter rule ids won't work because they don't change when the user changes the operator/value. It might be better to have the `haveFiltersChanged` state in the Provider and set to `true` when a filter is added/updated/removed.
@PriPatel do we want to have the save button disabled until the user changes something in the dashboard ? We only do superficial checks. E.g if we detect a user moved a tile we enable the button but we dont actually check if the tiles are in different positions compared to the original saved dashboard.